src/zapv2/__init__.py

```diff
@@ -110,25 +114,22 @@ def urlopen(self, *args, **kwargs):
            - `args`:  all non-keyword arguments.
            - `kwargs`: all other keyword arguments.
         """
-        kwargs['proxies'] = self.__proxies
-        return urllib.urlopen(*args, **kwargs).read()
+        return self.session.get(*args, proxies=self.__proxies, **kwargs).text
 
-    def _request_api(self, url, get=None):
+    def _request_api(self, url, query=None):
         """
         Shortcut for an API request. Will always add the apikey (if defined)
 
         :Parameters:
            - `url`: the url to GET at.
         """
-        session = requests.Session()
-        if self.__apikey is not None:
```

Comment:
@jotes - with this change I consistently get errors like:

> Traceback (most recent call last):
>   File "test.py", line 28, in <module>
>     print 'Spider progress %: ' + zap.spider.status(scanid)
>   File "build/bdist.linux-x86_64/egg/zapv2/spider.py", line 31, in status
>   File "build/bdist.linux-x86_64/egg/zapv2/__init__.py", line 156, in _request
>   File "build/bdist.linux-x86_64/egg/zapv2/__init__.py", line 146, in _request_api
>   File "/usr/lib/python2.7/site-packages/requests/sessions.py", line 477, in get
>     return self.request('GET', url, **kwargs)
>   File "/usr/lib/python2.7/site-packages/requests/sessions.py", line 465, in request
>     resp = self.send(prep, **send_kwargs)
>   File "/usr/lib/python2.7/site-packages/requests/sessions.py", line 573, in send
>     r = adapter.send(request, **kwargs)
>   File "/usr/lib/python2.7/site-packages/requests/adapters.py", line 424, in send
>     raise ConnectionError(e, request=request)
> requests.exceptions.ConnectionError: HTTPConnectionPool(host='127.0.0.1', port=8090): Max retries exceeded with url: http://zap/JSON/spider/view/status/?scanId=3&apikey=nb7idcoksh4gc6b8uatfi9hsj7 (Caused by ProxyError('Cannot connect to proxy.', error(104, 'Connection reset by peer')))
> 

This happens when polling the API checking for the spider status.
It works fine when I use a new session each time.